app/services/katello/event_queue.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  class EventQueue
+    @event_types = {}
+
+    def self.clear_events(event_type, object_id, on_or_earlier_than)
+      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
+    end
+
+    def self.peek_event
```

Comment:
`pop_event` ? I realize this does not remove it from the queue but it is this queues act of popping the top one off.

![popop](https://cloud.githubusercontent.com/assets/293363/16088301/78e208d8-32f4-11e6-8915-046920371bbc.gif)